[docs] Update font and styles

jonsamp opened the pull request:
# Why

Updates the design of docs to be more readable, and to better match expo.io. This includes updating the typeface and many styles.

# Changes

- Replaces Maison with Inter.
- Makes the docs full screen-width. I also constrained the content area to 1200px for large screens.
- Updates block quotes with a new design. Removes border-left design and makes it into a rounded box.
- Updates the left sidebar design. Removes the dashed line, adds styles to make titles and subtitles align correctly. Also updates the selected item style (now there's a dot next to the item you're on).
- Updates the scroll bars with new colors. They're now more subtle and round.
- Updates button design for "Try this example on Snack" buttons.
- Updates design of SDK selector under API reference.
- Replace emojis in the platform compatibility table with icons.
- Updates the permalink icon.
- Updates the external link icon.
- Makes border-radii in the navigation bar 4px.
- Fixes a responsive issue, where between two breakpoints it's impossible to see the sidebar list. 

# Screenshot

![Group 1](https://user-images.githubusercontent.com/6455018/92336315-3e90cd00-f06d-11ea-8ce2-2b1ff881bc14.png)
<img width="1713" alt="Screen Shot 2020-09-06 at 6 24 15 PM" src="https://user-images.githubusercontent.com/6455018/92336411-36855d00-f06e-11ea-9dee-8ef957707545.png">
<img width="1713" alt="Screen Shot 2020-09-06 at 6 24 24 PM" src="https://user-images.githubusercontent.com/6455018/92336413-371df380-f06e-11ea-9ff8-788f7a104cb9.png">
<img width="2668" alt="Screen Shot 2020-09-06 at 6 26 57 PM" src="https://user-images.githubusercontent.com/6455018/92336442-9419a980-f06e-11ea-9ace-186c10dba7b5.png">


# Test Plan

You should pull this PR and test it! It affects every page of the docs. I'm looking for styles that are now broken that are not broken on production.

I'm also looking for general design feedback!

jonsamp committed:
WIP

jonsamp committed:
many style updates

jonsamp committed:
adds background color

jonsamp committed:
adds content wrapper

jonsamp committed:
new link icon

jonsamp committed:
more style adjustments

jonsamp committed:
updates side bar styles

jonsamp committed:
better compatibilty table

jonsamp committed:
self-feedback

jonsamp committed:
feedback

jonsamp committed:
add bullet back

jonsamp committed:
fix code within h2

jonsamp committed:
responsive fixes

jonsamp committed:
back to a div for now

brentvatne commented:
this is a huge improvement to readability of the docs! just comparing the same page side by side my eyes are drawn to the new version of it. well done!

## the subheading here, "Essentials", doesn't look like it's nested under "Bare workflow"

<img width="282" alt="Screen Shot 2020-09-08 at 10 52 32 AM" src="https://user-images.githubusercontent.com/90494/92510968-8163b980-f1c1-11ea-8447-1141ecd34d44.png">


## some emojis are being rendered differently

<img width="742" alt="Screen Shot 2020-09-08 at 11 00 01 AM" src="https://user-images.githubusercontent.com/90494/92511646-74939580-f1c2-11ea-80ab-d28542786e08.png">

<img width="809" alt="Screen Shot 2020-09-08 at 11 00 12 AM" src="https://user-images.githubusercontent.com/90494/92511661-79584980-f1c2-11ea-9830-9fa01685b426.png">

<img width="1154" alt="Screen Shot 2020-09-08 at 11 03 13 AM" src="https://user-images.githubusercontent.com/90494/92511993-f97eaf00-f1c2-11ea-8aec-0033a0199201.png">

<img width="1135" alt="Screen Shot 2020-09-08 at 11 03 58 AM" src="https://user-images.githubusercontent.com/90494/92512004-ff749000-f1c2-11ea-9025-57273a9eb242.png">

## block quotes feel too similar to code blocks. before we distinguished them more clearly by removing the border except on the left side where there was a thicker border.

<img width="1130" alt="Screen Shot 2020-09-08 at 11 01 07 AM" src="https://user-images.githubusercontent.com/90494/92511737-9a209f00-f1c2-11ea-843d-9e1abfd8e05d.png">

<img width="1252" alt="Screen Shot 2020-09-08 at 11 01 15 AM" src="https://user-images.githubusercontent.com/90494/92511753-9f7de980-f1c2-11ea-9f43-5cb1479bf2a3.png">

## linebreak missing, unsure of cause

<img width="1164" alt="Screen Shot 2020-09-08 at 11 05 32 AM" src="https://user-images.githubusercontent.com/90494/92512235-58442880-f1c3-11ea-9dcf-ffdfa7dedc69.png">

<img width="1193" alt="Screen Shot 2020-09-08 at 11 06 35 AM" src="https://user-images.githubusercontent.com/90494/92512247-5d08dc80-f1c3-11ea-9f72-6b7993ad2a2d.png">

tommybru commented on docs/components/DocumentationHeader.js:
```diff
@@ -54,39 +53,39 @@ const STYLES_NAV = css`
-  @media screen and (min-width: ${Constants.breakpoints.mobileStrict}) {
-    display: none;
+  @media screen and (max-width: ${Constants.breakpoints.mobile}) {
+    display: block;
```
```suggestion
    display: flex;
```
The searchbar on the mobile nav is not vertically centered. If we want to keep `display: block`, I can look at some other ways to refactor.

brentvatne reviewed (changes requested):
commented

tommybru commented on docs/components/base/typography.js:
```diff
@@ -0,0 +1,47 @@
+export const h1 = css`
+  font-family: ${Constants.fonts.bold};
+  color: ${Constants.colors.black90};
+  font-size: 52px;
```
```suggestion
  font-size: 42px;
```
52px seemed a bit too large on desktop and mobile -- 42px felt more comfortable

jonsamp replied:
I made the text match the website, which adheres to a font scale. We can scale down this font stack for the docs if you both think it's too large. I'd just need to scale down all the headers, and then consider if we want to scale down the body text (which is at 16px right now). 

Here's two examples in both directions:
Example of large headers: https://reactnative.dev/docs/intro-react
Example of small headers: https://stripe.com/docs/api

To me, the larger sizes are inviting and make it easier to scan the headers. With that, if it's too much let's scale 'em back.  (cc @brentvatne @tommybru )

tommybru replied:
I'd be open to scaling down just the headers and seeing how the docs look. I think the 16px body text works really well. But I find the H1 a little too large to easily scan.

tommybru reviewed (approved):
Changes look great and very sleek to me. I love the new header with these updates. 

Left comments on small fixes, but am excited for this to merge.

cruzach reviewed (changes requested):
this is great! 🥇 making the docs full-width really opens up more room for us to add header links on the right for the individual page you're on (in the future, not this PR of course 😅 )

I have just a few suggestions:

- I think the code text size is too large in comparison to the regular paragraph text now, so either the paragraph text should be bumped up, or the code style text brought down:
![image](https://user-images.githubusercontent.com/35579283/92604158-30fe6180-f27e-11ea-90c5-f1e58a528018.png)
- same for the float-over element text size, it should probably match the regular paragraph text size:
![image](https://user-images.githubusercontent.com/35579283/92604297-5be8b580-f27e-11ea-8ac2-d7b171b82480.png)
- regarding the code blocks & block quotes brent mentioned- I say we leave the codeblocks as they are, and then block quotes should look less "separated" from the text that's surrounding them. Gmail does a really nice job with block quotes, just an indent and a left border:
![image](https://user-images.githubusercontent.com/35579283/92605189-6d7e8d00-f27f-11ea-81f9-dd5e02f21282.png)

jonsamp committed:
fix blockquotes

jonsamp committed:
fix font stacks

jonsamp committed:
Feedback incorporation

jonsamp commented:
@brentvatne  @cruzach:

## Blockquotes
I changed the blockquote to have a border left style, and also to align the text of the callout with the rest of the text. This serves two purposes: 1. the text flows more since it's all left aligned, 2. the breakout is more of a callout since the container is to the left of everything else. On mobile, I made the container of the block quote align with the surrounding text. Here's a screenshot of a code block and a blockquote:

<img width="1286" alt="Screen Shot 2020-09-09 at 11 48 03 AM" src="https://user-images.githubusercontent.com/6455018/92625030-310a5b80-f296-11ea-82ee-df8885d7345e.png">

## Sidebar
To make the left side bar have better visual hierarchy, I aligned the subtitles with the group titles. The hope is that this structure gives a clear hierarchy of the content.

<img width="542" alt="Screen Shot 2020-09-09 at 11 53 17 AM" src="https://user-images.githubusercontent.com/6455018/92625313-7d559b80-f296-11ea-8929-ea9ad9c3d263.png">

## Inline code
I changed the inline code style. It's now a little smaller, rounder, and now has a border. I also made it work inside block quotes. You can see both below:

<img width="997" alt="Screen Shot 2020-09-09 at 12 04 51 PM" src="https://user-images.githubusercontent.com/6455018/92625422-9d855a80-f296-11ea-947c-23be336a47a1.png">

## Emojis
Inter displays some emojis differently. My theory is that it has some emojis built-in and then displays them as black/white. I wasn't about to figure out how to fix this without modifying the font file itself, which id like to avoid doing since it's bad for the end-user (they'd have to download our custom Inter, even if they had it cached already). So I changed some emojis, like these arrows:

<img width="997" alt="Screen Shot 2020-09-09 at 12 14 56 PM" src="https://user-images.githubusercontent.com/6455018/92625476-b0982a80-f296-11ea-953e-06fc6a17757e.png">

One emoji that I couldn't get back to normal is the ⚠️ sign:

<img width="993" alt="Screen Shot 2020-09-09 at 12 15 36 PM" src="https://user-images.githubusercontent.com/6455018/92625644-eccb8b00-f296-11ea-80c6-9f155774e017.png">

## Tips
I also made the popovers have more reasonable text size/spacing:

<img width="938" alt="Screen Shot 2020-09-09 at 12 14 19 PM" src="https://user-images.githubusercontent.com/6455018/92625660-f1903f00-f296-11ea-840a-c6adae757f32.png">

## Other things:
- I added a fuller font-stack for all fonts. 
- I changed all our code fonts to our mono font stack.

jonsamp committed:
master installation

jonsamp committed:
updates snapshot

jonsamp committed:
lessen sidebar text size

jonsamp committed:
style tweaks

jonsamp committed:
fix terminal block comments

jonsamp committed:
fixes bad imports

jonsamp committed:
fix header

jonsamp committed:
fixes snapshot

cruzach commented:
Hm, personally I think the block quotes should be indented so that they are positioned "inside" the surrounding text, rather than aligned with it. This helps to separate the the vital information from notes and little extra bits of info, which is what we usually use block quotes for in the docs

brentvatne commented:
I like the new blockquote styling but I prefer it without the negative margin :)

<hr />

I noticed that spacing with emoji seems off with inter maybe?

![Screen Shot 2020-09-09 at 11 22 24 AM](https://user-images.githubusercontent.com/90494/92638329-f3ee9b00-f28e-11ea-9357-bda075e71366.png)

it looks like there is no space between the emoji and the text that follows

jonsamp committed:
blockquote inline, header size smaller

jonsamp committed:
center active state

jonsamp committed:
snapshot

jonsamp committed:
parse out emojis from block quotes

jonsamp committed:
snapshot

jonsamp committed:
fix inline buttons

jonsamp committed:
fix inline code wrapping

jonsamp committed:
better code sizing

jonsamp committed:
fix emoji

jonsamp committed:
fix header on learning more page

jonsamp committed:
better spacing around text divs

jonsamp committed:
snapshot

jonsamp committed:
fix app.json

jonsamp committed:
update summary details styles

jonsamp committed:
use demi

jonsamp committed:
snapshot

jonsamp committed:
restore prev styles on summary

jonsamp committed:
better table of contents styles

jonsamp committed:
back to orig h4 style

jonsamp committed:
better nesting levels

cruzach reviewed (commented):
I really like these changes, thank you so much @jonsamp !! 

My final nitpick- I'm not sure if this was changed in one of the recent commits or if I just noticed it this last check, but I think the sidebar page titles are a little bit closer together now. This doesn't really matter when looking at the API sidebar at https://docs.expo.io/versions/latest/ , but it's more noticeable at https://docs.expo.io/guides/, and for me makes it harder to read through them / find something I'm looking for

<img width="480" alt="docs" src="https://user-images.githubusercontent.com/35579283/92808312-a4d96080-f389-11ea-8064-8e5b03eab405.png">

jonsamp committed:
more space around links

jonsamp committed:
a littler tighter

jonsamp committed:
better terminal block line height

jonsamp committed:
fix a typo

cruzach reviewed (approved):
awesome, that little bump in space made _such_ an impact on readability

For me, this is ready to 🚀

brentvatne reviewed (approved):
:shipit:
